File: solutions/fsharp/log-levels/1/LogLevels.fs
module LogLevels

open System.Text.RegularExpressions

let pattern = 
    Regex("\[(\w+)\]:\s+(.*)")

let message (logLine: string): string = 
    pattern.Match(logLine).Groups[2].Value.Trim()

let logLevel(logLine: string): string = 
    pattern.Match(logLine).Groups[1].Value.ToLower()

let reformat(logLine: string): string =
    let lvl = logLevel logLine
    let msg = message logLine
    sprintf $"{msg} ({lvl})"

File: solutions/csharp/log-levels/1/LogLevels.cs
using System;

static class LogLine
{
    public static string Message(string logLine) =>
        logLine.Split(':')[1].Trim();

    public static string LogLevel(string logLine)
    {
        var firstLetterIndex = logLine.IndexOf('[') + 1;
        var rightBracketIndex = logLine.IndexOf(']');
        var length = rightBracketIndex - firstLetterIndex;
        return logLine.Substring(firstLetterIndex, length).ToLower();
    }

    public static string Reformat(string logLine)
    {
        var level = LogLine.LogLevel(logLine);
        var message = LogLine.Message(logLine);
        return $"{message} ({level})";
    }
}
